File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    
    let day = 0;
    
    let finish = [];
    
    while(progresses.length !== 0){
        //작업이 남아 있지 않을 때까지
        day++; //하루지남   
        
        for(let i = 0; i < progresses.length; i++) {
            progresses[i] += speeds[i];
        }
               
        let count = 0; //끝낼 일
    
        //첫타자가 완료된다면
        while(progresses[0] >= 100) {
            progresses.shift();
            speeds.shift();
            count++;
        }
        finish.push(count);
    }
     
    return finish.filter((progress) => progress !== 0);
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.io.*;
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        List<Integer> list = new ArrayList<>();
        
        int idx = 0; // 아직 배포 안 된 가장 앞 기능
        
        while(idx < progresses.length) {
            
            for(int i = 0; i < progresses.length; i++) {
                progresses[i] += speeds[i];
            }
            
            // 앞에서부터 완료된 기능 확인
            int deployCount = 0;
            int now = idx;
            
            for(int i = now; i < progresses.length; i++) {
                if(progresses[i] >= 100) {
                    deployCount++;
                    idx++;
                } else break;
            }
            
            if(deployCount > 0) list.add(deployCount);
        }
        
        int[] answer = new int[list.size()];
        
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}